### DISCUSSION TITLE
How (if possible) to have /api as a page

### DISCUSSION THREAD

**Discussion by amirrsyd** [vercel/next.js#16055] [Help]
We would like to have a page that is accessed at domain.com/api. As the /api folder in /pages is reserved for API routes, I'm wondering id there is a way to make this happen. Choosing not to have any API routes and having an api.js does not seem to work.
```
pages/
├── api.js
``` 
Is having an /api page not possible at all with Next or is there a way to make this possible, has anybody does something like this?

**Comment by jamesmosier** [comment]
You could use the [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) functionality in your `next.config.js`:

```js
module.exports = {
  async rewrites() {
    return [
      {
        source: '/api',
        destination: '/foo', // pages/foo.js
      },
    ];
  },
};
```
